diff -r 0cbab3d7640d bin/spider.py
--- a/bin/spider.py	Sun Apr 25 09:34:12 2010 +0400
+++ b/bin/spider.py	Sun Apr 25 10:01:11 2010 +0400
@@ -8,6 +8,7 @@
 
 import psycopg2
 import sys
+import subprocess
 import string
 import re
 import socket
@@ -232,13 +233,12 @@
     hoststr = sharestr(proto, host, port)
     savepath = share_save_path(proto, host, port)
     patchmode = tree_id != None and os.path.isfile(savepath)
-    address = socket.gethostbyname(host)
+    address = host
     log("Scanning %s (%s) ...", (hoststr, address))
     start = datetime.datetime.now()
-    if patchmode:
-        data = run_scanner(command, address, proto, port, "-u " + quote_for_shell(savepath))
-    else:
-        data = run_scanner(command, address, proto, port)
+    patchmode = False
+    data = subprocess.Popen("type %(s)s" % {'s': quote_for_shell(savepath)},
+        shell = True, stdin=subprocess.PIPE, stdout=subprocess.PIPE, universal_newlines=True)
     save = tempfile.TemporaryFile(bufsize=-1)
     line_count = 0
     for line in data.stdout:
