diff -r c3d1f1bfd05a bin/spider.py
--- a/bin/spider.py	Mon Apr 26 03:58:31 2010 +0400
+++ b/bin/spider.py	Mon Apr 26 04:31:29 2010 +0400
@@ -13,7 +13,9 @@
 import socket
 import tempfile
 import os
+import os.path
 import sys
+import subprocess
 import traceback
 import shutil
 import datetime
@@ -223,15 +225,14 @@
 def scan_share(db, share_id, proto, host, port, tree_id, command):
     cursor = db.cursor()
     hoststr = sharestr(proto, host, port)
-    savepath = share_save_path(proto, host, port)
+    savepath = os.path.join('dump', share_save_str(proto, host, port))
     patchmode = tree_id != None and os.path.isfile(savepath)
-    address = socket.gethostbyname(host)
+    address = host
     log("Scanning %s (%s) ...", (hoststr, address))
     start = datetime.datetime.now()
-    if patchmode:
-        data = run_scanner(command, address, proto, port, "-u " + quote_for_shell(savepath))
-    else:
-        data = run_scanner(command, address, proto, port)
+    patchmode = False
+    data = subprocess.Popen("type %(s)s" % {'s': quote_for_shell(savepath)},
+        shell = True, stdin=subprocess.PIPE, stdout=subprocess.PIPE, universal_newlines=True)
     save = tempfile.TemporaryFile(bufsize=-1)
     line_count = 0
     line_count_patch = 0
@@ -288,7 +289,7 @@
         log("Failed to save contents of %s to file %s.", (hoststr, savepath))
     save.close()
     cursor.execute("""
-        UPDATE shares SET tree_id = %(t)s, last_scan = now()  WHERE share_id = %(s)s;
+        UPDATE shares SET tree_id = %(t)s WHERE share_id = %(s)s;
         """, {'s': share_id, 't': tree_id})
     if qcache.totalsize >= 0:
         cursor.execute("""
